### PR TITLE
new: Migrate to Nimbus. Update to latest TypeScript, Babel, ESLint, and more.

### PR DESCRIPTION
**Pull request by milesj**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

Migrates to Nimbus, which bumps us up to the latest versions of Babel, ESLint, TypeScript, Jest, etc. Because of this, I had a lot of type issues and lint failures to fix. The biggest change was `any` to `unknown`.

## Motivation and Context

Standardizes our tools with other projects.

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

File: packages/app-shell/src/components/AppContext.ts [hunk 1]
```diff
@@ -1,4 +1,20 @@
 import React from 'react';
 import { Context } from '../types';
 
-export default React.createContext<Context | null>(null);
+const noop = () => '';
+
+export default React.createContext<Context>({
```
Comment (milesj) [inline packages/app-shell/src/components/AppContext.ts]: Add defaults to it's easier for consumers to integrate.

File: packages/core/src/components/DataTable/index.tsx [hunk 2]
```diff
@@ -214,11 +216,13 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
...
       const { changeLog }: { changeLog: ChangeLog } = this.state;
       const { originalIndex } = row.rowData.metadata;
-      if (Object.prototype.hasOwnProperty.call(changeLog, originalIndex)) {
-        changeLog[originalIndex][key] = newVal;
+
+      if (changeLog[originalIndex] && changeLog[originalIndex][key]) {
+        changeLog[originalIndex][key] = { newVal };
       } else {
-        changeLog[originalIndex] = { [key]: newVal };
+        changeLog[originalIndex] = { [key]: { newVal } };
```
Comment (milesj) [inline packages/core/src/components/DataTable/index.tsx]: The types had it as `{ newVal: string }` but the code was just `newVal`. Not sure which is correct.

**Comment by milesj** [comment]
@milesj @stefhatcher @hayes @alecklandgraf This is ready. You can ignore the tests for the most part, since it was mainly type/lint fixes. The source is the same, but I did make a few adjustments. I'm not sure how some of these type changes will affect consumers.

**Comment by alecklandgraf** [comment]
Is it possible to get a prerelease to test?

File: packages/app-shell/src/components/Toasts/PopToast.tsx [hunk 3]
```diff
@@ -1,30 +1,29 @@
...
     } else if (props.success) {
-      context.addSuccessToast(message, props);
+      addSuccessToast(message, props);
     } else {
-      context.addInfoToast(message, props);
+      addInfoToast(message, props);
     }
-  }, [ref.current]);
+
+    // We only care when the label changes, not all the props
```
Comment (stefhatcher) [inline packages/app-shell/src/components/Toasts/PopToast.tsx]: `label` -> `message`

File: packages/core/src/components/Menu/Separator.tsx [hunk 4]
```diff
@@ -7,7 +7,7 @@ export type Props = {
    * a property existed in this type. I have no idea why.
```
Comment (stefhatcher) [inline packages/core/src/components/Menu/Separator.tsx]: lol

**Pull request merged**